### PR TITLE
add DB backup functionality

### PR DESCRIPTION
**Pull request by HariHive**
- [ ] Add a link to the ticket in the PR title or add a description of work in the PR title
- [ ] dashcam firmware was loaded on a device and successfully runs
- [ ] semantic version was incremented inside `src/config/index.ts`

### CODE REVIEW

File: python/backup.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import sqlite3
+import os
+import sys
+
+if len(sys.argv) != 2:
```
Comment (roger-berman) [inline python/backup.py:5]: Nit: Can you use argparse instead?
As well as can you put this under a main.
Example of both here: https://github.com/Hivemapper/sensor_fusion/blob/main/sensor_fusion_service/main.py#L269

File: python/backup.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    print("Usage: python backup_script.py <source_db>")
+    sys.exit(1)
+source_db = sys.argv[1]
+backup_db = '/data/recording/backup.db'
```
Comment (roger-berman) [inline python/backup.py:9]: Nit: Can you move this constant to the top and capitalize the entire variable?

File: src/util/runCommands.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { exec } from 'child_process';
+import { promisify } from 'util';
+
+const asyncExec = promisify(exec);
```
Comment (roger-berman) [inline src/util/runCommands.ts:4]: Does this type of functionality not already exist somewhere int the code? If not then this is totally fine, just wanna make sure we don't have a bunch of code duplication.

File: src/util/runCommands.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+export const runSequence = async(cmds: string[]) => {
+    try{
+        for await (const cmd of cmds) {
```
Comment (roger-berman) [inline src/util/runCommands.ts:8]: Does this need to be a for await loop? You already await the asyncExec within the loop?

**Review by roger-berman — COMMENTED** [review]
Looks good just a few comments

File: src/util/runCommands.ts (hunk 3 above)
Comment (HariHive) [reply src/util/runCommands.ts:4]: I have checked everywhere else looks like promisified exec is used, please lmk if you want me to change it

File: src/util/runCommands.ts (hunk 4 above)
Comment (HariHive) [reply src/util/runCommands.ts:8]: taken this inspiration from src/util/index.ts and src/routes/instrumentation.ts please lmk if you need me to change

Comment (mazar-farran) [reply src/util/runCommands.ts:8]: I think the `for await` is not needed, you can change to `for`

**Comment by mazar-farran** [comment]
Would this work?

```
// @ts-ignore
let backup = db.backup('backup.db', (err: any) => {
    backup.step(-1, (err: any) => {
        backup.finish(function(err: any) {
            console.log('Backup finished', err);
        });
    });
});
```

There seems to be a backup api in node-sqlite3 but it's not part of the TypeScript interface.  https://github.com/TryGhost/node-sqlite3/blob/master/lib/sqlite3.js#L105

It's kind of clunky but it might be better than calling python

This worked locally for me on laptop but I haven't tested on HDC

**Pull request merged**